arbor/morph/sample_tree.cpp

```diff
@@ -28,8 +28,15 @@ void sample_tree::reserve(msize_t n) {
 }
 
 msize_t sample_tree::append(msize_t p, const msample& s) {
-    if ((empty() && p!=mnpos) || p>=size()) {
-        if (p!=mnpos) throw invalid_sample_parent(p, size());
+    // Assert that the root (first) sample in a tree has mnpos as its parent.
```

Comment:
This fixes a nasty bug (seg fault) caused when trying to insert a sample with `mnpos` as its parent when it wasn't a root.